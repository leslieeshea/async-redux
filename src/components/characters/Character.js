import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Character({ character }) {
  return (
    <section>
      <Link to={`/redux-avatar-promise/${character.id}`}>
        <img src={character.img} />
        <p>{character.name}</p>
      </Link>
    </section>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
  }).isRequired
};

export default Character;
