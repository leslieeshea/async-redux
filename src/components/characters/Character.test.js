import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const wrapper = shallow(<Character character={{ id: '1', name: 'aang', img: '/img' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
