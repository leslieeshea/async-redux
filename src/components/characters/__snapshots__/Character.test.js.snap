// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<section>
  <Link
    to="/1"
  >
    <img
      src="/img"
    />
    <p>
      aang
    </p>
  </Link>
</section>
`;
